tests/test_tokenization_common.py

```diff
@@ -540,6 +540,52 @@ def test_tokenizers_common_properties(self):
                 for attr in attributes_list:
                     self.assertTrue(hasattr(tokenizer, attr))
 
+    def test_tokenizers_common_ids_setters(self):
+        tokenizers = self.get_tokenizers()
+        for tokenizer in tokenizers:
+            with self.subTest(f"{tokenizer.__class__.__name__}"):
+                attributes_list = [
+                    "bos_token",
+                    "eos_token",
+                    "unk_token",
+                    "sep_token",
+                    "pad_token",
+                    "cls_token",
+                    "mask_token",
+                ]
+
+                try:
+                    vocab = tokenizer.get_vocab()
+                except NotImplementedError:
+                    vocab = None
+                should_test_setters_not_none = (vocab is not None) and (len(vocab) > 0)
```

Comment:
This is obviously up for discussion, but I wonder if this case is not only designed for `ByT5`. If that's the case, I think it's worth removing this complexity from the common test and designing a test specifically for ByT5, what do you think? :smile:

Reply:
Yes, I've checked it with a script and `get_vocab` is only missing in `ByT5` and `canine` tokenizers, but this test is already "disabled" for canine.

My thought process was basically:
1. I need to get one random token id to test the setters, let's get the first item of the vocab.
2. It looks that, at least, `ByT5` needs this check. Because I'm not sure I'm not missing anything I'll take the safe path and always make sure that we have a vocab.

I was thinking a pro of my approach is that if a new model doesn't define `get_vocab`, the author doesn't need to make any decision regarding this test. However, I understand one could argue that this isn't a pro actually. I mean, if someone adds a new model that doesn't define `get_vocab`, the test will fail and the author will be aware of this situation and will need to explicitly override or disable (define `pass` empty body) the test. Is this better? Now I'm leaning to think it is, so OK, I will modify the PR as you suggest :)

Reply:
Yes, I've checked it with a script and `get_vocab` is only missing in `ByT5` and `canine` tokenizers, but this test is already "disabled" for canine.

My thought process was basically:
1. I need to get one random token id to test the setters, let's get the first item of the vocab.
2. It looks that, at least, `ByT5` needs this check. Because I'm not sure I'm not missing anything I'll take the safe path and always make sure that we have a vocab.

I was thinking a pro of my approach is that if a new model doesn't define `get_vocab`, the author doesn't need to make any decision regarding this test. However, I understand one could argue that this isn't a pro actually. I mean, if someone adds a new model that doesn't define `get_vocab`, the test will fail and the author will be aware of this situation and will need to explicitly override or disable (define `pass` empty body) the test. Is this better? Now I'm leaning to think it is, so OK, I will modify the PR as you suggest :)

Reply:
Indeed, I completely understand your way of thinking. `Canine` and `ByT5` are 2 tokenizers a bit different from the others so I think it's good to do it that way! Thanks a lot for your last commit!

This discussion makes me think that it would be nice to have a test for `Canine` as well, because this tokenizer also defines special tokens. 
https://github.com/huggingface/transformers/blob/70851a6bf0bf0cf39cb4919c1d3fd66affe9f0db/src/transformers/models/canine/tokenization_canine.py#L79-L90

What do you think about it? :smile:

Reply:
Ah yes, I know nothing about `Canine` , and after seeing the `SPECIAL_CODEPOINTS` I thought about it like a fixed config for special tokens, but I guess that if the setters are there (inherited and not disabled/raise not implemented), they should be tested.

Reply:
Thank you very much for the new test! It is perfect!

I should have mentioned it at the same time, it seems to me that V also defines special tokens: 

https://github.com/huggingface/transformers/blob/f7196f2e63b14e9fbb4ad664e71912aab3b484cf/src/transformers/models/speech_to_text/tokenization_speech_to_text.py#L109-L123

Do you think we can also make a custom test for it?

Reply:
😄 I could have realized too. It was a remainder of the original test that used the token id 0 to test the setters and a problem that I wrongly attributed to the existence of a fixed vocab 😅. Using `get_vocab`, the default test was already able to cover this tokenizer.